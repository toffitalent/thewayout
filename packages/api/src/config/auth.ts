import { env } from '@two/config';
import { UserType } from '@two/shared';

export const SCOPES = ['user'];

// Roles CANNOT be requested! Roles are automatically expanded to include all scopes specified.
// IMPORTANT: Ensure role names are UNIQUE and DO NOT CONFLICT with scope names!!!
export const ROLES = {
  admin: [UserType.Client, UserType.Employer, UserType.Rsp, ...SCOPES],
  self: [...SCOPES],

  [UserType.Client]: [],
  [UserType.Employer]: [],
  [UserType.Rsp]: [],
};

export const auth = () => {
  const auth = {
    algorithm: 'HS256',
    audience: 'two-api',
    issuer: 'two',
    secret: env({
      development:
        'MGICAQACEQC0GLtyi4/+XKuxzUliJ0B1AgMBAAECECyOaeG+8ZMGQ6jVVls5GIECCQDrhO4++DizMQIJAMPB/yr+OqiFAgg4tVU4TEQ6wQIIdGJ+ht0nmtECCQDHw8QHR1TrFA==',
      test: 'MGICAQACEQC0GLtyi4/+XKuxzUliJ0B1AgMBAAECECyOaeG+8ZMGQ6jVVls5GIECCQDrhO4++DizMQIJAMPB/yr+OqiFAgg4tVU4TEQ6wQIIdGJ+ht0nmtECCQDHw8QHR1TrFA==',
      env: 'API_AUTH_SECRET',
    }),
    ttl: 900,

    scopes: SCOPES,
    roles: ROLES,
  };

  const authMiddleware = {
    algorithm: auth.algorithm,
    audience: auth.audience,
    issuer: auth.issuer,
    roles: auth.roles,
    secret: auth.secret,
  };

  return { auth, server: { auth: authMiddleware } };
};
