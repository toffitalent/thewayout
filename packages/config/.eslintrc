{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/config/tsconfig.json"]
      }
    }
  },
  "rules": {
    "global-require": "off",
    "import/no-dynamic-require": "off"
  }
}
