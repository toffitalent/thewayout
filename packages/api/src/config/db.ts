import { env, register } from '@two/config';

export const db = register('db', () => ({
  hostname: env('API_DB_HOSTNAME', 'localhost'),
  username: env('API_DB_USERNAME', 'two'),
  password: env('API_DB_PASSWORD', ''),
  database: env('API_DB_DATABASE', 'two'),
  debug: env('API_DB_DEBUG', false),
  port: env('API_DB_PORT', 5432),
}));
