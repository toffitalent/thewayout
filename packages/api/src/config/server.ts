import { env, register } from '@two/config';

export const server = register('server', () => ({
  port: env({
    default: 3000,
    env: 'PORT',
    prefix: false,
  }),
  proxy: true,

  log: env({
    default: {
      level: 'info',
    },
    development: {
      level: 'debug',
      colors: true,
    },
    test: {
      level: 'none',
    },
  }),

  store: {
    url: env({
      default: 'redis://localhost:6379',
      env: 'API_REDIS_URL',
    }),
  },

  // Middleware
  auth: {
    //
    // NOTE: These settings are overridden in the "auth" config file
    //
    // enabled: true,
    // algorithm: 'HS256',
    // roles: [],
    // secret: '',
  },
  cache: {
    // enabled: true,
    // maxAge: 0,
  },
  cors: {
    // enabled: true,
    // origin: '',
    // allowHeaders: [],
    // allowMethods: [],
    // exposeHeaders: [],
    // maxAge: number,
  },
  csp: {
    enabled: false,
    // directives: {},
  },
  cto: {
    // enabled: true,
    // directive: '',
  },
  gzip: {
    // enabled: true
  },
  hsts: env({
    default: {
      // enabled: true,
      // maxAge: 0,
      // includeSubdomains: true,
      // preload: false,
    },
    development: {
      enabled: false,
    },
    staging: {
      maxAge: 3600,
    },
    production: {
      maxAge: 31536000,
    },
  }),
  json: {
    // enabled: true,
  },
  logger: env({
    default: {
      // enabled: true,
      // format: ':method :url :status :response-time ms - :length',
    },
    test: {
      enabled: false,
    },
    staging: {
      format:
        ':id :ip :app :user [:date] ":method :url" :status :length :response-time ":referrer" ":user-agent"',
    },
    production: {
      format:
        ':id :ip :app :user [:date] ":method :url" :status :length :response-time ":referrer" ":user-agent"',
    },
  }),
  methodOverride: {
    // enabled: true,
  },
  parser: {
    // enabled: true,
    // encoding = 'utf-8',
    // formLimit = '56kb',
    // json = true,
    // jsonLimit = '1mb',
    // multipart = false,
    // parsedMethods = ['POST', 'PUT', 'PATCH'],
    // text = false,
    // textLimit = '56kb',
    // urlencoded = true,
  },
  referrer: {
    // enabled: true,
    // directive: ['origin-when-cross-origin', 'strict-origin-when-cross-origin'],
  },
  requestId: {
    // enabled: true,
  },
  responseTime: env({
    default: {
      // enabled: true,
    },
    production: {
      enabled: false,
    },
  }),
  throttle: env({
    default: {
      // enabled: true,
      // duration: 900,
      // max: 450,
      // maxAuth: 450,
      // maxUnauth: 450,
    },
    test: {
      max: 10000,
    },
  }),
  version: {
    // enabled: true,
    availableVersions: [1],
  },
  xframe: {
    // enabled: true,
    // directive: 'deny',
  },
  xss: {
    // enabled: true,
    // mode: 'block',
  },
}));
