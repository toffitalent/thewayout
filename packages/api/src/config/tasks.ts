import { env, register } from '@two/config';

export const tasks = register('tasks', () => ({
  queue: {
    endpoint: env('AWS_ENDPOINT_URL'),
    region: env({
      default: 'us-west-2',
      env: 'AWS_REGION',
    }),
    url: env({
      default: '',
      test: '__QUEUE_URL__',
      env: 'API_TASKS_QUEUE_URL',
    }),
  },
}));
