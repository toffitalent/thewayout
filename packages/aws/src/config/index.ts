import { env, Manager } from '@disruptive-labs/config';
import { App } from 'aws-cdk-lib';
import { URL } from 'url';
import { secrets } from './secrets';

const DEPLOY = ['mail', 'web'];
let appEnv: string;

export const getDeployConfig = (context = '') => {
  const deploy = context.split(',').map((value) => value.toLowerCase().trim());

  return DEPLOY.reduce(
    (config, item) => ({
      ...config,
      [item]: deploy.includes('true') || deploy.includes(item),
    }),
    {},
  );
};

export const getEnvId = (id: string) =>
  `${appEnv.replace(/^\w/, (char) => char.toUpperCase())}-${id}`;

export const getHostname = (url?: string) => {
  if (!url) throw new Error('Missing required URL config');
  return new URL(url).hostname;
};

export const loadConfig = (app: App): Manager => {
  appEnv = app.node.tryGetContext('env')?.toLowerCase();

  if (!appEnv) {
    throw new Error(
      'Context env variable missing. Specify environment using the context option, e.g. `-c env=production`',
    );
  }

  if (process.env.APP_ENV && process.env.APP_ENV !== appEnv) {
    throw new Error(
      `Env mismatch! APP_ENV is set to "${process.env.APP_ENV}" but env context is "${appEnv}"`,
    );
  }

  // Set APP_ENV before loading main config
  process.env.APP_ENV = appEnv;
  process.env.NODE_ENV = process.env.NODE_ENV || appEnv;

  // eslint-disable-next-line global-require
  const { default: config, load } = require('@two/config');

  load('aws', [
    () => ({
      account: env('AWS_ACCOUNT', '114149725394'),
      region: env('AWS_REGION', 'us-west-2'),
      deploy: getDeployConfig(app.node.tryGetContext('deploy')),
      // Convert all configured URLs to hostnames
      hostnames: Object.keys(config.get('urls', {})).reduce(
        (urls, key) => ({
          ...urls,
          [key]: getHostname(config.get(`urls.${key}`)),
        }),
        {},
      ),
      secrets,
    }),
  ]);

  return config;
};

export type Config = ReturnType<typeof loadConfig>;
