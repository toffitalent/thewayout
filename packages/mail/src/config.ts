import { env, load } from '@two/config';

const mail = () => ({
  queue: {
    endpoint: env('AWS_ENDPOINT_URL'),
    region: env({
      default: 'us-west-2',
      env: 'AWS_REGION',
    }),
    url: env({
      default: '',
      test: '__QUEUE_URL__',
      env: 'MAIL_QUEUE_URL',
    }),
  },
  transport: env({
    development: 'smtp',
    test: 'json',
    staging: 'json',
    production: 'ses',
  }),
  sendgrid: {
    apiKey: env('SENDGRID_API_KEY', ''),
  },
  smtp: {
    host: env('MAIL_SMTP_HOST', 'localhost'),
    port: env('MAIL_SMTP_PORT', 1025),
  },
});

export const mailConfig = load('mail', [mail]);
