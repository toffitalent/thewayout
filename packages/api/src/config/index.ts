import { load } from '@two/config';
import { auth } from './auth';
import { db } from './db';
import { server } from './server';
import { tasks } from './tasks';

export const apiConfig = load('api', [auth, db, server, tasks]);

export default apiConfig;
