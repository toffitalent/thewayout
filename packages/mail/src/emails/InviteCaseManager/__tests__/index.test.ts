import config from '@two/config';
import { renderTemplate } from '@test';
import { InviteCaseManager } from '..';

describe('InviteCaseManager', () => {
  const context = {
    firstName: 'First',
    lastName: 'Last',
    email: 'test@test.com',
    phone: '1112223333',
    invitationId: 'INVITATION_TEST_ID',
    rspName: 'Organization Name',
  };

  const email = new InviteCaseManager(context);

  test('augments invitation url context', () => {
    expect(email.getContext()).toEqual({
      ...context,
      config,
      url: `${config.get(
        'urls.web',
      )}/ssp/signup/?firstName=First&lastName=Last&email=test%40test.com&phone=1112223333&invitationId=INVITATION_TEST_ID&rspName=Organization%20Name`,
    });
  });

  test('renders correctly', async () => {
    expect(await email.render()).toMatchSnapshot();
    expect(renderTemplate('InviteCaseManager', 'html', email.getContext())).toMatchSnapshot();
    expect(renderTemplate('InviteCaseManager', 'subject', email.getContext())).toMatchSnapshot();
    expect(renderTemplate('InviteCaseManager', 'text', email.getContext())).toMatchSnapshot();
  });
});
