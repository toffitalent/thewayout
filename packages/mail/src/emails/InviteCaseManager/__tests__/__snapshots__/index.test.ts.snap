// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteCaseManager renders correctly 1`] = `
{
  "html": "pug-test-stub",
  "subject": "pug-test-stub",
  "text": "pug-test-stub",
}
`;

exports[`InviteCaseManager renders correctly 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta name="viewport" content="width=device-width, initial-scale=1.0"/><meta name="x-apple-disable-message-reformatting"/><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="color-scheme" content="light dark"/><meta name="supported-color-schemes" content="light dark"/><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nunito:400,700&amp;display=swap"/><link rel="stylesheet" href="@app/styles/index.scss" data-inline="true"/><!--[if mso]><style type="text/css">.f-fallback {
  font-family: Arial, sans-serif;
}</style><![endif]--></head><body><span class="preheader"></span><table class="email__wrapper" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td align="center"><table class="email__content" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="email__masthead"><a class="f-fallback email__masthead__logo" href="http://__WEB__"><img src="@app/assets/logo.png" alt="The Way Out"/></a></td></tr><tr><td class="email__body" width="570" cellpadding="0" cellspacing="0"><table class="email__body__inner" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="content"><div class="f-fallback"><h1>Hi First,</h1><p>Your company admin has invited you to create a case manager account for Organization Name on The Way Out.</p><table class="body__action" align="center" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td align="center"><table width="100%" border="0" cellspacing="0" cellpadding="0" role="presentation"><tr><td align="center"><a class="f-fallback button button--primary" href="http://__WEB__/ssp/signup/?firstName=First&amp;lastName=Last&amp;email=test%40test.com&amp;phone=1112223333&amp;invitationId=INVITATION_TEST_ID&amp;rspName=Organization%20Name" target="_blank">Sign Up</a></td></tr></table></td></tr></table><table class="body__sub" role="presentation"><tr><td><p class="f-fallback sub">If you&apos;re having trouble with the button above, copy and paste the URL below into your web browser.</p><p class="f-fallback sub"><a class="nocolor" href="http://__WEB__/ssp/signup/?firstName=First&amp;lastName=Last&amp;email=test%40test.com&amp;phone=1112223333&amp;invitationId=INVITATION_TEST_ID&amp;rspName=Organization%20Name">http://__WEB__/ssp/signup/?firstName=First&lastName=Last&email=test%40test.com&phone=1112223333&invitationId=INVITATION_TEST_ID&rspName=Organization%20Name</a></p></td></tr></table></div></td></tr></table></td></tr><tr><td><table class="email__footer" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="content" align="center"><p class="f-fallback sub align-center">&copy; 2024 The Way Out. All rights reserved.</p><p class="f-fallback sub align-center">The Way Out, Inc.<br/><a href="#">PO Box 70620</a><br/><a href="#">Milwaukee, WI 53207</a></p></td></tr></table></td></tr></table></td></tr></table></body></html>"
`;

exports[`InviteCaseManager renders correctly 3`] = `"Invitation to join Organization Name"`;

exports[`InviteCaseManager renders correctly 4`] = `
"Hi First,

Your company admin has invited you to create a case manager account for Organization Name on The Way Out.

Sign Up: http://__WEB__/ssp/signup/?firstName=First&lastName=Last&email=test%40test.com&phone=1112223333&invitationId=INVITATION_TEST_ID&rspName=Organization%20Name

--
Â© 2024 The Way Out. All rights reserved.

The Way Out, Inc.
PO Box 70620
Milwaukee, WI 53207"
`;
